daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/CommandsVersion.scala

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.engine
+
+import com.digitalasset.daml.lf.transaction.VersionTimeline
+import com.digitalasset.daml.lf.language.{LanguageVersion, LanguageMajorVersion => LMV}
+import com.digitalasset.daml.lf.data.Ref._
+import com.digitalasset.daml.lf.command._
+import com.digitalasset.daml.lf.data.{ImmArray, ImmArrayCons}
+import com.digitalasset.daml.lf.language.Ast.Package
+
+import scala.annotation.tailrec
+
+/** Some DAML-LF operations (curently contract key lookup and fetch) are parametrized
+  * by the language version that "originated" the transactions.
+  *
```

Comment:
I think it is a bad idea to parametrized the interpreter by daml-lf version.

I will let this consideration in the deserialization layer, and use for instance the `feature flags` concept.
(for instance, for versions < x, the unserialization set the feature flag to on, and for version >= s to off).

- This introduces unecessary complexity in the core of the system that is already very complex. 
- It is simpler to reason about a set of featur  flags that well established and defined in the unique place, than a set of feature distributed in the code (and not so cleary identified) parametrize by  version.
- This opens the pandora box to parametrize everything by language versions.
- you put the language version, in the module level exactly in the same place as the feature flag.

Reply:
@remyhaemmerle-da I agree that it is undesirable, however:

* I disagree that it's "unnecessary". How do you propose to deal with cases like the one in this pr?
* I do not understand the last point. Could you clarify?

Reply:
I've taked to @remyhaemmerle-da and agreed that it's better to parametrize execution by a flag, rather than by a version. I've implemented this refactor in 06e76885b6931c27de1e633330c2dcf9666955fd .